File: BookStoreApi/Database/Models/Author.cs
using Dapper.Contrib.Extensions;

namespace BookStoreApi.Database.Models
{
    [Table("Authors")]
    public class Author
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}

File: BookStoreFront/src/app/models/author.ts
export interface Author {
  id: number;
  name: string;
  description: string;
  createdAt?: Date;
  updatedAt?: Date;
}

export interface APaginationInfo {
  totalCount: number;
  pageSize: number;
  pageNumber: number;
  totalPages: number;
}

export interface AuthorListResponse {
  authors?: Author[];
  pagination?: APaginationInfo;
}

// create-author-request.ts
export interface CreateAuthorRequest {
  name: string;
  description: string;
}

// update-author-request.ts
export interface UpdateAuthorRequest {
  name: string;
  description: string;
}